magic/fetcher_internal.py

```diff
@@ -1,39 +1,16 @@
-import csv
+
 import json
 import os
 import shutil
 import urllib.request
 import zipfile
 from email.utils import formatdate
 
-import pkg_resources
 import requests
 
-import configuration
-import database
-
-def legal_cards(force=False):
-    resource_id = 'legal_cards'
-    if force:
-        resource_id = None
-    return fetch('http://pdmtgo.com/legal_cards.txt', 'utf-8', resource_id).strip().split('\n')
-
-def version():
-    return pkg_resources.parse_version(json.loads(fetch('https://mtgjson.com/json/version.json')))
-
-def mtgo_status():
-    try:
-        return json.loads(fetch('https://magic.wizards.com/sites/all/modules/custom/wiz_services/mtgo_status.php'))['status']
-    except (FetchException, json.decoder.JSONDecodeError):
-        return 'UNKNOWN'
-
-def all_cards():
-    s = unzip('https://mtgjson.com/json/AllCards-x.json.zip', 'AllCards-x.json')
-    return json.loads(s)
+from magic import database
 
-def all_sets():
-    s = unzip('https://mtgjson.com/json/AllSets.json.zip', 'AllSets.json')
-    return json.loads(s)
+SESSION = requests.Session()
```

Comment:
These module-level variables that are instantiated when the module is imported freak me out a little bit. For a while we were double fetching whatsinstandard on startup because of the stuff I added in rotation. I don't know python standards well enough to know if I should get over my squeamishness or respond to it. It's sort of like a member variable or a static member variable but it's getting instantiated at import time which I find weird. I can probably cope.

Why is it good to have a Session? Some kind of caching?